example/tests/example_test.py

```diff
@@ -1,22 +1,60 @@
 import unittest
 
-from snapstack import Runner
+from snapstack import Plan, Step, Setup, Cleanup
 
 
 class ExampleTest(unittest.TestCase):
 
-    def test(self):
-        r = Runner(
+    def test_example(self):
+        '''
+        _test_example_
+
+        Basic test, which installs and tests and example
+        snap, after letting the Plan object build the default
+        snapstack.
+
+        Note: this will install snapstack on your system!
+
+        '''
+        example = Step(
             snap='example',  # Name of the snap
-            location='./tests/',  # Parent location for tests and files
-                                  # below. {local} is a magic string that
-                                  # points at the 'tests' directory.
+            script_loc='./tests/',  # Parent location for tests and files
+                                    # below. {local} is a magic string that
+                                    # points at the 'tests' directory.
             tests=['example.sh'],  # Test the snap
             files=['example.json'],  # Config files
-            base=None  # Don't override default Base.
+            snap_store=False  # Build from local source, rather than
+                              # install from the snap store.
         )
+        plan = Plan(tests=[example])
+        plan()
+
+    def test_custom_base(self):
+        '''
+        _test_custom_base_
+
+        Reference code for adding and removing stuff from our Setup
+        and Cleanup.
+
+        '''
+        setup = Setup()
+        cleanup = Cleanup()
+
+        # Add a step.
+        example = Step(
+            snap='example',
+            script_loc='./tests/',
+            tests=['example.sh'],
+            files=['example.json'],
+        )
+        setup.add_steps(('example', example))
+
+        # Remove steps.
+        setup.remove_steps(
+            'keystone', 'nova', 'neutron', 'glance', 'nova_hypervisor',
+            'neutron_ext_net')
 
-        try:
-            r.run()
-        finally:
-            r.cleanup()
+        # Run with our changes (this will download the config files,
+        # but then install nothing but the example snap).
+        plan = Plan(tests=[], setup=setup.steps(), cleanup=cleanup.steps())
```

Comment:
@coreycb Hmmm ... it seems like that would add a lot of complexity, for no gain. The Setup and Cleanup classes are just shortcuts for creating a list of Steps. If you want to override all of the steps in the base setup and cleanup, you could simply skip using the Setup and Cleanup classes altogether, and pass in your own lists. (I can change the check to None so that an empty list works.)

 Similarly, if you want to avoid using the "add steps" and "remove steps" methods, you could pass in something like:

```
    Plan(setup=Setup().steps() + [some_step, some_other_step] ...)
```

Do the Setup and Cleanup classes add too much complexity? I could go back to just using a list, possibly parsed from a JSON file. I wanted to make it easy to say "skip the keystone step", without messing around with parsing the list. But if they're making the code harder to grok, maybe they should go.